pkg/cli/debug/debug.go

```diff
@@ -594,6 +594,30 @@ func (o *DebugOptions) getContainerImageCommand(pod *corev1.Pod, container *core
 func (o *DebugOptions) transformPodForDebug(annotations map[string]string) (*corev1.Pod, []string) {
 	pod := o.Attach.Pod
 
+	if len(o.ToNamespace) > 0 {
+		svcAcctOrig := pod.Spec.ServiceAccountName
+		volumesOrig := pod.Spec.Volumes
+		clearVolumes(pod)
+		if len(pod.Spec.ServiceAccountName) != 0 {
+			pod.Spec.ServiceAccountName = ""
+		}
+		if len(pod.Spec.DeprecatedServiceAccount) != 0 {
+			pod.Spec.DeprecatedServiceAccount = ""
+		}
+		if len(pod.Spec.InitContainers) != 0 {
+			klog.V(4).Infof("PodSpec InitContainers cleared to run with '--to-namespace', run without '--to-namespace' to keep InitContainers")
```

Comment:
Why initcontainers?

Reply:
I empirically found it to be true, didn't pinpoint why yet - I'll get back to you on that.

Reply:
I see why now, most initContainers are configured to run commands that depend on VolumeMounts, so if not debugging the initContainer, that cmd will try to run as pod starts, then will leave debug pod in error state w/ PodInitializing.  Now we'll clear InitContainers only if not explicitly debugging the initContainer (if you pass -c initContainerName, though, the command will become `/bin/sh` in the debug pod, so that will not fail when debugging initContainers.